src/Symfony/Component/HttpFoundation/UriSigner.php

```diff
@@ -89,6 +93,41 @@ public function sign(string $uri/* , \DateTimeInterface|\DateInterval|int|null $
      */
     public function check(string $uri): bool
     {
+        try {
+            $this->verify($uri);
+        } catch (SignedUriException) {
```

Comment:
shouldn't we avoid using exceptions for signaling?
maybe some internal return code would be better?
or an extra $throw argument to a private doVerify method?

Reply:
> shouldn't we avoid using exceptions for signaling?

I do this a lot so I'm curious why

Reply:
exceptions are super heavy: object creation + stack trace collection.

Reply:
I see. Done.